pkg/event/module.go

```diff
@@ -154,28 +159,108 @@ func NewEventsDBModule(db DBEvents) EventsDBModule {
 
 func (cm *eventsDBModule) Store(_ context.Context, m map[string]string) error {
 
-	// prepare the query to insert the event in the DB
-	origin := "keycloak" // for the moment only events of Keycloak
-	realmName := m["realmId"]
-	agentUserID := ""   // no agents, only events from Keycloak
-	agentUsername := "" // no agents, only events from Keycloak
-	userID := m["userId"]
-	username := m["username"]
-	ctEventType := ""
-	kcEventType := m["type"]
-	kcOperationType := m["operationType"]
-	clientID := m["clienId"]
-	additionalInfo := "" // all the rest in a JSON?
-	/*if m["type"] == "LOGIN_ERROR" {
-		fmt.Println("The rest of event:")
-		fmt.Println(m)
-	}*/
-
-	_, err := cm.db.Exec(insertEvent, origin, realmName, agentUserID, agentUsername, userID, username, ctEventType, kcEventType, kcOperationType, clientID, additionalInfo)
+	// if ctEventType is not "", then record the events in MariaDB
+	if m["ct_event_type"] != "" {
+		origin := "keycloak" // for the moment only eventss of Keycloak
+		realmName := m["realmId"]
+		//agentUserID - userId of who is performing an action
+		agentUserID := m["userId"]
+		//agentUsername - username of who is performing an action
+		agentUsername := m["username"] // normally, username can be found in the details key of the map
+		//agentRealmName - realm of who is performing an action
+		agentRealmName := "" // found in authdetails
+		//userID - ID of the user that is impacted by the action
+		userID := "" // found in authdetails, in resourcePath
+		//username - username of the user that is impacted by the action
+		username := ""                    //
+		ctEventType := m["ct_event_type"] // the ct event type is established before at the component level
+		kcEventType := m["type"]
+		kcOperationType := m["operationType"]
+		clientID := m["clientId"]
 
-	if err != nil {
-		//TODO: how is this error treated further?
-		return err
+		//userId is in the resourcePath
+		if resourcePath, ok := m["resourcePath"]; ok {
+			reg := regexp.MustCompile(`[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}`)
+			if strings.HasPrefix(resourcePath, "users") {
```

Comment:
Is it possible to simplify it with? :
if m["resourcePath"] == "users"{
//....
}

Reply:
No, because resourcePath has as preffix "users" , followed by an UUID. It can also be "roles" followed also by an UUID